tools/offline_log_viewer/topic_manifest.py

```diff
@@ -0,0 +1,80 @@
+from typing import Any
+from reader import Reader
+from controller import read_topic_config
+
+
+def to_compression_str(value: int) -> str:
+    return ['none', 'gzip', 'snappy', 'lz4', 'zstd'][value]
+
+
+def to_cleanup_policy_bitflags_str(value: int) -> str:
+    return ['none', 'delete', 'compact', 'compact,delete'][value]
+
+
+def to_compaction_strategy_str(value: int) -> str:
+    return ['offset', 'timestamp', 'header'][value]
+
+
+def to_timestamp_type_str(value: int) -> str:
+    return ['CreateTime', 'LogAppendTime'][value]
+
+
+def to_opt_json(value: Any | None, converter=lambda x: x) -> Any | None:
+    return converter(value) if value is not None else None
+
+
+def to_tristate_json(key: str,
+                     value: dict,
+                     converter=lambda x: x) -> dict[str, Any | None]:
+    if 'value' in value:
+        return {key: converter(value['value'])}
+    if value['state'] == 'empty':
+        return {key: None}
+    return {}
+
+
+def decode_topic_manifest(path: str) -> dict[str, Any]:
+    return Reader(open(path, "rb")).read_envelope(
+        lambda rdr, _: {
+            'cfg': rdr.read_envelope(read_topic_config, max_version=1),
+            'initial_revision': rdr.read_int64()
+        })
+
+
```

Comment:
part of the followup ticket: to move decoding to rp_storage_tool, 
and to update test code to accept the schema of the new topic manifest,

https://github.com/redpanda-data/redpanda/issues/17502

> I think you mentioned this offline, but it's a little surprising for this to show up in the offline log viewer, given topic manifests aren't in any log. I guess there isn't really a better place for it though...

Yeah, I tried to hack into rp_storage_tool, but in the interest of minimizing issues, I went with offline_log_viewer since I was already familiar with it, and it already had 90% of the code done.

but it's a temp measure, we should have this in rp_storage_tool

> It also feels a bit odd to bake semantics into the tool. Did you consider having the tool just dump out all of its JSON, and then having the test code pull out just the dictionary it needs?

For the same reason as above, since the tests are in Python and all the schema is implicit, I went with translating the new manifest in the old format. In the end, a lot of commits are just to make sure that tests can find topic_manifest.bin. I'm not sure yet how many commits are needed to update the code to use the new schema. This too, will be part of the follow-up.